BinTree/BinTree.cpp

```diff
@@ -84,6 +213,57 @@ std::ostream& operator<<(std::ostream& os, const BinTree& p)
             os << *p.right;
         }
     }
-    
     return os;
 }
+
+void BinTree::displaySideways()
+{
+    cout << endl;
+    displaySideways("");
+}
+
+void BinTree::displaySideways(string buff)
+{
+    size_t offset = 1;
+    if (root) {
+        offset += root->getDate().length();
+    }
+    string spacer = string(offset + buff.length(), ' ');
+    if (right) {
+        right->displaySideways(spacer);
+    }
+    if (root) {
+        if (right) {
+            cout << string(spacer.length() - 1, ' ') << '/' << endl;
+        }
+        cout << buff;
+        cout << *root << endl;
+        if (left) {
+            cout << string(spacer.length() - 1, ' ') << '\\' << endl;
+        }
+    }
+    if (left) {
+        left->displaySideways(spacer);
+    }
+}
+
+int BinTree::getHeight(const NodeData &node) const
+{
+    if (*root == node) {
+        return getHeight();
```

Comment:
When finding the requested node, getHeight() will let us know how far this node is from the leaf (or if this is the leaf).